### PR TITLE
feat: run A/B test for delayed MarTech stack

### PR DESCRIPTION
**Pull request by ramboz**
## Changes

- Track experiment data in the data layer so we can track business KPIs in Analytics and drill down per variant
- Move all the MarTech to the `third-party.js` and toggle eager/delayed loading based on the experiment variant that is served
- Properly `await` all async code in the block decoration to avoid leaking CPU cycles that would clutter the time between lazy and delayed
- Parallelizing key resources to reduce the delay in the critical path for the LCP, and delaying those that aren't needed

## Test URLs
- Before: https://main--petplace--hlxsites.hlx.live/
- After:
  - No MarTech: https://delayed-martech--petplace--hlxsites.hlx.live/experiments/delayed-martech/?martech=off
  - Eager MarTech: https://delayed-martech--petplace--hlxsites.hlx.live/experiments/delayed-martech/?experiment=delayed-martech/control (i.e. same as the before page essentially)
  - Delayed MarTech: https://delayed-martech--petplace--hlxsites.hlx.live/experiments/delayed-martech/?experiment=delayed-martech/challenger-1

### Original Page
Screenshot 2024-04-11 at 14 10 51

### Experiment

| Disabled MarTech | Eager MarTech | Delayed MarTech |
|-|-|-|
| Screenshot 2024-04-11 at 14 10 11 |Screenshot 2024-04-11 at 14 11 06|Screenshot 2024-04-11 at 14 10 40|

Key observations:
- the delayed MarTech only adds 0.1s delay to the FCP on top of the reference page with the MarTech disabled, so the main content isn't noticeably delayed. On the other hand, the main content is shown >60% faster as the original page, and the first content is shown >30% faster
- the delayed MarTech has the same SI as the original page, so while we delayed some code, the overall page experience is stable after the same duration on both sides
- the TBT is also down so the page will be more responsive for the end users while it loads (i.e. scrolling around will be easier and touch interactions have less delay)

### CODE REVIEW

File: templates/tag-index/tag-index.js [hunk 1]
```diff
@@ -11,7 +11,6 @@ import {
   meterCalls,
 } from '../../scripts/scripts.js';
 import { pushToDataLayer } from '../../scripts/utils/helpers.js';
-import { adsenseFunc } from '../../scripts/adsense.js';
```
Comment (ramboz) [inline templates/tag-index/tag-index.js:14]: Delay the `adsense.js` until the lazy phase where it is actually used

File: templates/category-index/category-index.js [hunk 2]
```diff
@@ -12,7 +12,6 @@ import {
   isTablet,
   meterCalls,
 } from '../../scripts/scripts.js';
-import { adsenseFunc } from '../../scripts/adsense.js';
```
Comment (ramboz) [inline templates/category-index/category-index.js:15]: Delay the `adsense.js` until the lazy phase where it is actually used

File: templates/breed-page/breed-page.js [hunk 3]
```diff
@@ -1,4 +1,3 @@
-import { adsenseFunc } from '../../scripts/adsense.js';
```
Comment (ramboz) [inline templates/breed-page/breed-page.js:1]: Delay the `adsense.js` until the lazy phase where it is actually used

File: templates/breed-index/breed-index.js [hunk 4]
```diff
@@ -1,4 +1,3 @@
-import { adsenseFunc } from '../../scripts/adsense.js';
```
Comment (ramboz) [inline templates/breed-index/breed-index.js:1]: Delay the `adsense.js` until the lazy phase where it is actually used

File: templates/article-page/article-page.js [hunk 5]
```diff
@@ -9,7 +9,6 @@ import {
   createBreadCrumbs,
   getCategories,
 } from '../../scripts/scripts.js';
-import { adsenseFunc } from '../../scripts/adsense.js';
```
Comment (ramboz) [inline templates/article-page/article-page.js:12]: Delay the `adsense.js` until the lazy phase where it is actually used

File: blocks/footer/footer.js [hunk 6]
```diff
@@ -21,16 +21,16 @@ export default async function decorate(block) {
     const footer = document.createElement('div');
     footer.innerHTML = html;
 
-    decorateIcons(footer);
+    await decorateIcons(footer);
```
Comment (ramboz) [inline blocks/footer/footer.js:24]: Properly `await` the async operations so they don't leak into the delayed phase timeout

File: head.html [hunk 7]
```diff
@@ -22,18 +22,3 @@
     'analytics_storage': 'denied'
   });
 </script>
-<script>
```
Comment (ramboz) [inline head.html:25]: Moving this to the `third-party.js` instrumentation so we can control if it is loaded and when (eager vs lazy)

File: scripts/datalayer.js [hunk 8]
```diff
@@ -120,6 +121,15 @@ const handleCtaClicks = () => {
   }
 };
 
+const handleExperiments = () => {
```
Comment (ramboz) [inline scripts/datalayer.js:124]: Push experiment metadata to the data layer so we can track it in Analytics

File: scripts/scripts.js [hunk 9]
```diff
@@ -1,4 +1,3 @@
-import globRegex from './glob-regex/glob-regex.js';
```
Comment (ramboz) [inline scripts/scripts.js:1]: Defer loading until it is actually used when the newsletter popup gets rendered

File: scripts/scripts.js [hunk 10]
```diff
@@ -20,7 +19,6 @@ import {
   toClassName,
   waitForLCP,
 } from './lib-franklin.js';
-import { login, logout, acquireToken, isLoggedIn } from './lib/msal/msal-authentication.js'
```
Comment (ramboz) [inline scripts/scripts.js:23]: This isn't actually used in this file, so removing the extra dependency

File: scripts/scripts.js [hunk 11]
```diff
@@ -82,7 +80,7 @@ window.hlx.plugins.add('experimentation', {
 window.hlx.plugins.add('martech', {
   url: './third-party.js',
   condition: () => new URLSearchParams(window.location.search).get('martech') !== 'off',
-  load: 'lazy',
+  load: 'eager',
```
Comment (ramboz) [inline scripts/scripts.js:83]: Switch to `eager` loading so we can handle the case where we run the martech early in the A/B test.

File: scripts/scripts.js [hunk 12]
```diff
@@ -898,7 +896,11 @@ export async function getConfiguration(sheet) {
 }
 
 async function addNewsletterPopup() {
-  const newsletterConfig = await getConfiguration('newsletter-popup');
+  const [globRegex, newsletterConfig] = await Promise.all([
```
Comment (ramboz) [inline scripts/scripts.js:899]: Load dependencies in parallel

File: scripts/third-party.js [hunk 13]
```diff
@@ -1,29 +1,7 @@
-import { loadScript } from './lib-franklin.js';
+import { getMetadata, loadScript } from './lib-franklin.js';
 // eslint-disable-next-line import/no-cycle
 import { isMobile } from './scripts.js';
 
-function loadMSClarity() {
```
Comment (ramboz) [inline scripts/third-party.js:5]: Simplify the JS to a simple `loadScript`

File: scripts/third-party.js [hunk 14]
```diff
@@ -64,14 +70,23 @@ export async function loadLazy() {
...
-    loadScript('https://securepubads.g.doubleclick.net/tag/js/gpt.js', {
-      async: '',
-    });
+    loadScript('https://securepubads.g.doubleclick.net/tag/js/gpt.js', { async: true });
   }
 }
 
 export function loadDelayed() {
-  loadMSClarity();
+  if (window.location.hostname === 'www.petplace.com'
```
Comment (ramboz) [inline scripts/third-party.js:78]: Move this up so it renders before the rest of the delayed martech is handled (i.e. prioritize people with needing accessibility adjustments over the marketing stack)

**Review by ankyc-phi — APPROVED** [review]
looks great

**Pull request merged**